cli/lang-generator.js

```diff
@@ -0,0 +1,91 @@
+const fs = require('fs');
+const path = require('path');
+const sergeDirectories = require('../core.serge.json');
+
+const outputRoot = path.join(__dirname, '../generated');
+const outputPath = path.join(outputRoot, 'lang');
+
+function alphaSort(a, b) {
+	a = a.toLowerCase();
+	b = b.toLowerCase();
+	if (a < b) {
+		return -1;
+	} else if (a > b) {
+		return 1;
+	}
+	return 0;
+}
+
+function cleanDir(dirPath) {
+	if (!fs.existsSync(dirPath)) {
+		return;
+	}
+	const files = fs.readdirSync(dirPath);
+	files.forEach((file) => {
+		const currentPath = path.join(dirPath, file);
+		if (fs.lstatSync(currentPath).isDirectory()) {
+			cleanDir(currentPath);
+		} else {
+			fs.unlinkSync(currentPath);
+		}
+	});
+
+	fs.rmdirSync(dirPath);
+
+}
+function appendComponent(sergeComponent) {
+	const directory = `${sergeComponent.source_dir}/`;
+	const filenames = fs.readdirSync(directory);
+
+	if (!filenames) {
+		throw 'dir is empty';
+	}
+
+	filenames.forEach((filename) => {
+		const filepath = directory + filename;
+		const lang = path.basename(filepath, '.json');
+
+		const terms = {};
+		const data = fs.readFileSync(filepath);
+		if (data) {
+			const currentLangTerms = JSON.parse(data);
+			Object.keys(currentLangTerms)
+				.sort(alphaSort)
+				.forEach((key) => {
+					terms[key] = currentLangTerms[key] && currentLangTerms[key].translation;
+				});
+		}
+
+		const component = sergeComponent.name;
+		appendLangFile(lang, component, terms);
+	});
+}
+function appendLangFile(lang, component, terms) {
+	const generateFilePath = path.join(outputPath, `${lang}.js`);
+
+	if (!fs.existsSync(generateFilePath)) {
+		fs.writeFileSync(generateFilePath, '');
+	}
+	const json = JSON.stringify(terms);
+	fs.appendFileSync(generateFilePath, `export const ${component} = JSON.parse('${json}');\n`, 'utf8', (err) => {
+		if (err) throw err;
+	});
+}
+function generate() {
+	if (!fs.existsSync(outputRoot)) {
+		fs.mkdirSync(outputRoot, {recursive: true});
+	}
+	cleanDir(outputPath);
+	fs.mkdirSync(outputPath, {recursive: true});
+
+	sergeDirectories.forEach(appendComponent);
+}
+
+try {
+	generate();
```

Comment:
If you think some additional logs would be useful, feel free to include them.  We can also use `chalk` if we wish.  While colors in Travis don't always show up in logs, they can be nice locally.

Reply:
Cool! added